File: 2018-jpa-java/src/main/java/org/scripton/domain/repository/UserRepository.java
package org.scripton.domain.repository;

import org.scripton.domain.model.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

/**
 * @author archmagece
 * @date 2017-05-12
 */
@Repository
public interface UserRepository extends JpaRepository<UserEntity, Long> {
	UserEntity findByUsername(String username);
}

File: 2021-rabbitmq-jpa/domain/jpa/src/main/kotlin/sample/domain/jpa/repository/UserRepository.kt
package sample.domain.jpa.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import sample.domain.jpa.model.UserEntity
import java.util.*

@Repository
interface UserRepository : JpaRepository<UserEntity, Long> {
    fun findByUsername(username: String): Optional<UserEntity>
    fun findByEmail(email: String): Optional<UserEntity>
}
